org.alloytools.alloy.application/src/main/java/edu/mit/csail/sdg/alloy4whole/SimpleGUI.java

```diff
@@ -1842,9 +1851,10 @@ public final String compute(final Object input) throws Exception {
 				Expr e = CompUtil.parseOneExpression_fromString(root, str);
 				if ("yes".equals(System.getProperty("debug")) && VerbosityPref.get() == Verbosity.FULLDEBUG) {
 					SimInstance simInst = convert(root, ans);
-					return simInst.visitThis(e).toString() + (simInst.wasOverflow() ? " (OF)" : "");
-				} else
-					return ans.eval(e).toString();
+					if (simInst.wasOverflow())
+						return simInst.visitThis(e).toString() + " (OF)";
+				}
+				return ans.eval(e);
```

Comment:
I suggest reverting this change (lines 1854--1857), because `if () { return; } else { return; }` is more structured than `if () { if () return } return;`: the former has 2 possible code paths, and the later has 3.  

Also, did you intentionally change the semantics of this block for the case when "debug" is true and there was no overflow?  In that case, this code wastefully computes `simInst = convert(root, ans)` and then returns `ans.eval(e)`, whereas the previous code would return `simInst.visitThis(e).toString()`.

Reply:
This is pretty messy code. It always returned a String and I wanted the original SimTupleSet if possible. It also has a horrible API, allowing it to called with Object and the interpreting what type it was. 

This code needs to be replaced and I am happy to do this if you file an issue. Not sure we should try to improve this code in this PR.

Reply:
I changed the flow because I needed the real object, not the string.So only when it was overflow, I would return a string because it contained the error. So I need an extra path to distinguish:

        debug => ( overflow => string-answer else real object answer ) else real-object-answer

So not sure how to prevent a third path. On the contrary, your suggestion would introduce a 4th path?